modules/services/gpg-agent.nix

```diff
@@ -9,24 +9,42 @@ let
 
   homedir = config.programs.gpg.homedir;
 
+  pinentryBinPath = optionalString cfg.pinentryFlavor != null
+    (if cfg.pinentryFlavor == "mac" then
```

Comment:
I went for this specific series of conditions in an attempt to prevent pulling in any unnecessary pinentry packages. Especially in the case of `pinentry_mac` which is only supported on Darwin.